### PR TITLE
fix(app): fix jump to current step when outside current window

### LINKED ISSUE #10898: 6.0 Feedback: Jump to current step does not always display
### Overview

_No response_

### Current Behavior

Jump to current step is not always displayed when a user is not active on the run lug. Have to scroll a few times for the jump to current step to display 

see video:

https://user-images.githubusercontent.com/30632549/175561392-5cbb5dc7-b59e-42b4-aa9e-dbb68cd99111.MOV

 

### Expected Behavior

Jump to current step should display after commands are outside of current window display 

### Steps To Reproduce

_No response_

### Operating system

Windows

### Robot setup or anything else?

_No response_

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

run log jump to current step was not working when the viewport was outside of the current command
window. this reuses the component logic on mount to jump to the current command window.

closes #10898

# Changelog

  - Fixes run log jump to current step when outside current command window

# Review requests

for long protocols, confirm that scrolling outside the current command window displays the 'jump to current step' button, and that clicking the button scrolls as expected.

# Risk assessment

low

### CODE REVIEW

**Comment by brenthagen** [comment]
@y3rsh this PR may affect start/end time issues in https://github.com/Opentrons/opentrons/issues/11047

File: app/src/organisms/Devices/ProtocolRun/RunLog.tsx [hunk 1]
```diff
@@ -48,8 +48,8 @@ import { StepItem } from './StepItem'
 import type { RunCommandSummary } from '@opentrons/api-client'
 import type { RunTimeCommand, CommandStatus } from '@opentrons/shared-data'
 
-const AVERAGE_ITEM_HEIGHT_PX = 52 // average px height of a command item
-const WINDOW_OVERLAP = 40 // number of command items that fall within two adjacent windows
+const AVERAGE_ITEM_HEIGHT_PX = 56 // average px height of a command item
+const WINDOW_OVERLAP = 42 // number of command items that fall within two adjacent windows
```
Comment (b-cooper) [inline app/src/organisms/Devices/ProtocolRun/RunLog.tsx]: why did the window overlap need updating?

File: app/src/organisms/Devices/ProtocolRun/RunLog.tsx [hunk 2]
```diff
@@ -345,11 +349,9 @@ export function RunLog({ robotName, runId }: RunLogProps): JSX.Element | null {
       transform="translate(-50%)"
       borderRadius={SPACING.spacing6}
       display={
-        isRunStarted && (isCurrentStepBelow || isCurrentStepAbove)
-          ? DISPLAY_FLEX
-          : DISPLAY_NONE
+        isCurrentStepBelow || isCurrentStepAbove ? DISPLAY_FLEX : DISPLAY_NONE
```
Comment (b-cooper) [inline app/src/organisms/Devices/ProtocolRun/RunLog.tsx]: we're not checking `isRunStarted` anymore, was it unnecessary?

**Review by b-cooper — APPROVED** [review]
The changes look good.  Just a couple of clarifying questions

File: app/src/organisms/Devices/ProtocolRun/RunLog.tsx (hunk 1 above)
Comment (brenthagen) [reply app/src/organisms/Devices/ProtocolRun/RunLog.tsx]: the change to `AVERAGE_ITEM_HEIGHT_PX ` affects `AVERAGE_WINDOW_HEIGHT_PX`, which affects the window index set on scroll. `42` worked for window overlap when I tested, but may not be the correct number

File: app/src/organisms/Devices/ProtocolRun/RunLog.tsx (hunk 2 above)
Comment (brenthagen) [reply app/src/organisms/Devices/ProtocolRun/RunLog.tsx]: `isRunStarted` was only checking for the existence of the current item ref - which was part of the problem with the "jump to current step" button not rendering. I'm thinking now, should we instead condition display on the run being the current run? i'm guessing we don't want "jump to current step" to display for historical runs

**Pull request merged**